sdk/src/types/block/semantic/error.rs

```diff
@@ -144,32 +237,71 @@ impl TryFrom<u8> for TransactionFailureReason {
 
     fn try_from(c: u8) -> Result<Self, Self::Error> {
         Ok(match c {
-            1 => Self::InputUtxoAlreadySpent,
-            2 => Self::ConflictingWithAnotherTx,
-            3 => Self::InvalidReferencedUtxo,
-            4 => Self::InvalidTransaction,
-            5 => Self::SumInputsOutputsAmountMismatch,
-            6 => Self::InvalidUnlockBlockSignature,
-            7 => Self::TimelockNotExpired,
-            8 => Self::InvalidNativeTokens,
-            9 => Self::StorageDepositReturnUnfulfilled,
-            10 => Self::InvalidInputUnlock,
-            11 => Self::SenderNotUnlocked,
-            12 => Self::InvalidChainStateTransition,
-            13 => Self::InvalidTransactionIssuingTime,
-            14 => Self::InvalidManaAmount,
-            15 => Self::InvalidBlockIssuanceCreditsAmount,
-            16 => Self::InvalidRewardContextInput,
-            17 => Self::InvalidCommitmentContextInput,
-            18 => Self::MissingStakingFeature,
-            19 => Self::FailedToClaimStakingReward,
-            20 => Self::FailedToClaimDelegationReward,
-            21 => Self::TransactionCapabilityNativeTokenBurningNotAllowed,
-            22 => Self::TransactionCapabilityManaBurningNotAllowed,
-            23 => Self::TransactionCapabilityAccountDestructionNotAllowed,
-            24 => Self::TransactionCapabilityAnchorDestructionNotAllowed,
-            25 => Self::TransactionCapabilityFoundryDestructionNotAllowed,
-            26 => Self::TransactionCapabilityNftDestructionNotAllowed,
+            0 => Self::None,
```

Comment:
Can't we just use https://docs.rs/strum/latest/strum/derive.FromRepr.html?

Reply:
Done

Reply:
Nice